src/server/string_family.cc

```diff
@@ -444,6 +444,17 @@ OpStatus SetCmd::SetExisting(const SetParams& params, PrimeIterator it, ExpireIt
   PrimeValue& prime_value = it->second;
   EngineShard* shard = op_args_.shard;
 
+  DbSlice& db_slice = shard->db_slice();
+
+  uint64_t at_ms;
+  if (params.keep_expire) {
+    at_ms = std::max((long unsigned int) 0, db_slice.ExpireTime(e_it) - op_args_.db_cntx.time_now_ms);
```

Comment:
it's not needed. if keep_expire is true why would you even call UpdateExpire below?